geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/BinAggregatingIterator.scala

```diff
@@ -40,190 +37,8 @@ import scala.util.control.NonFatal
  * Iterator that computes and aggregates 'bin' entries
  */
 class BinAggregatingIterator
-    extends KryoLazyAggregatingIterator[ByteBufferResult] with SamplingIterator with LazyLogging {
-
-  import BinAggregatingIterator._
-
-  var trackIndex: Int = -1
-  var geomIndex: Int = -1
-  var dtgIndex: Int = -1
-  var labelIndex: Int = -1
-
-  var getTrackId: (KryoBufferSimpleFeature) => Int = _
-  var getDtg: (KryoBufferSimpleFeature) => Long = _
-  var isDtgArray: Boolean = false
-  var linePointIndex: Int = -1
-
-  var binSize: Int = 16
-  var sort: Boolean = false
-
-  var sampling: Option[(SimpleFeature) => Boolean] = _
-
-  var writeBin: (KryoBufferSimpleFeature, ByteBufferResult) => Unit = _
-
-  override def init(options: Map[String, String]): ByteBufferResult = {
-    geomIndex = options(GEOM_OPT).toInt
-    labelIndex = options.get(LABEL_OPT).map(_.toInt).getOrElse(-1)
-
-    trackIndex = options(TRACK_OPT).toInt
-    getTrackId = if (trackIndex == -1) {
-      (sf) => sf.getID.hashCode
-    } else {
-      (sf) => {
-        val track = sf.getAttribute(trackIndex)
-        if (track == null) { 0 } else { track.hashCode() }
-      }
-    }
-    dtgIndex = options(DATE_OPT).toInt
-    isDtgArray = options.get(DATE_ARRAY_OPT).exists(_.toBoolean)
-    getDtg = if (dtgIndex == -1) {
-      (_) => 0L
-    } else if (isDtgArray) {
-      (sf) => {
-        try {
-          sf.getAttribute(dtgIndex).asInstanceOf[java.util.List[Date]].get(linePointIndex).getTime
-        } catch {
-          case _: IndexOutOfBoundsException => 0L
-        }
-      }
-    } else {
-      (sf) => sf.getDateAsLong(dtgIndex)
-    }
-
-    binSize = if (labelIndex == -1) 16 else 24
-    sort = options(SORT_OPT).toBoolean
-
-    sampling = sample(options)
-
-    // derive the bin values from the features
-    val writeGeom: (KryoBufferSimpleFeature, ByteBufferResult) => Unit = if (sft.isPoints) {
-      if (labelIndex == -1) writePoint else writePointWithLabel
-    } else if (sft.isLines) {
-      if (labelIndex == -1) writeLineString else writeLineStringWithLabel
-    } else {
-      if (labelIndex == -1) writeGeometry else writeGeometryWithLabel
-    }
-
-    writeBin = sampling match {
-      case None => writeGeom
-      case Some(samp) => (sf, bb) => if (samp(sf)) { writeGeom(sf, bb) }
-    }
-
-    val batchSize = options(BATCH_SIZE_OPT).toInt * binSize
-
-    val buffer = ByteBuffer.wrap(Array.ofDim(batchSize)).order(ByteOrder.LITTLE_ENDIAN)
-    val overflow = ByteBuffer.wrap(Array.ofDim(binSize * 16)).order(ByteOrder.LITTLE_ENDIAN)
-
-    new ByteBufferResult(buffer, overflow)
-  }
-
+    extends BaseAggregatingIterator[ByteBufferResult] with BinAggregatingUtils {
   override def notFull(result: ByteBufferResult): Boolean = result.buffer.position < result.buffer.limit
```

Comment:
can we move this to the common class? I think I have handled multiple batches in the coprocessor (although i'm not actually sure it matters in hbase since it won't return until the end anyway...)